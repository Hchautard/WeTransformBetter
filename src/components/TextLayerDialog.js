import React from 'react';
import { Dialog, DialogTitle, DialogContent, DialogActions, TextField, Button, 
         FormControl, InputLabel, Select, MenuItem, Slider } from '@mui/material';
import { ChromePicker } from 'react-color';

const TextLayerDialog = ({ 
  open, 
  onClose, 
  textInput, 
  setTextInput, 
  textFont, 
  setTextFont, 
  textSize, 
  setTextSize, 
  textColor, 
  setTextColor, 
  addTextLayer 
}) => {
  return (
    <Dialog open={open} onClose={onClose}>
      <DialogTitle>Add Text Layer</DialogTitle>
      <DialogContent>
        <TextField
          autoFocus
          margin="dense"
          label="Text"
          fullWidth
          value={textInput}
          onChange={(e) => setTextInput(e.target.value)}
        />
        <FormControl fullWidth margin="normal">
          <InputLabel>Font</InputLabel>
          <Select value={textFont} onChange={(e) => setTextFont(e.target.value)}>
            <MenuItem value="Arial">Arial</MenuItem>
            <MenuItem value="Times New Roman">Times New Roman</MenuItem>
            <MenuItem value="Courier New">Courier New</MenuItem>
            <MenuItem value="Verdana">Verdana</MenuItem>
          </Select>
        </FormControl>
        <Slider
          value={textSize}
          onChange={(e, newValue) => setTextSize(newValue)}
          min={12}
          max={1000}
          valueLabelDisplay="auto"
        />
        <ChromePicker
          color={textColor}
          onChangeComplete={(color) => setTextColor(color.hex)}
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose}>Cancel</Button>
        <Button onClick={addTextLayer} color="primary">Add Text</Button>
      </DialogActions>
    </Dialog>
  );
};

export default TextLayerDialog;
