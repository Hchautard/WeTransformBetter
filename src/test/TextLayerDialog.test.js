import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import TextLayerDialog from '../components/TextLayerDialog';

describe('TextLayerDialog Component', () => {
  let mockOnClose, mockSetTextInput, mockSetTextFont, mockSetTextSize, mockSetTextColor, mockAddTextLayer;

  beforeEach(() => {
    mockOnClose = jest.fn();
    mockSetTextInput = jest.fn();
    mockSetTextFont = jest.fn();
    mockSetTextSize = jest.fn();
    mockSetTextColor = jest.fn();
    mockAddTextLayer = jest.fn();

    render(
      <TextLayerDialog
        open={true}
        onClose={mockOnClose}
        textInput=""
        setTextInput={mockSetTextInput}
        textFont="Arial"
        setTextFont={mockSetTextFont}
        textSize={24}
        setTextSize={mockSetTextSize}
        textColor="#000000"
        setTextColor={mockSetTextColor}
        addTextLayer={mockAddTextLayer}
      />
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('allows the user to type in the text input', () => {
    const input = screen.getByLabelText('Text');
    fireEvent.change(input, { target: { value: 'Hello World' } });
    expect(mockSetTextInput).toHaveBeenCalledWith('Hello World');
  });

  test('allows the user to change the text size using the slider', () => {
    const slider = screen.getByRole('slider');
    fireEvent.change(slider, { target: { value: 48 } });
    expect(mockSetTextSize).toHaveBeenCalledWith(48);
  });

  test('allows the user to change the text color using ChromePicker', () => {
    const colorPicker = screen.getByText('Add Text Layer').parentElement.querySelector('input[type="color"]');
    if (colorPicker) {
      fireEvent.change(colorPicker, { target: { value: '#ff5733' } });
      expect(mockSetTextColor).toHaveBeenCalledWith('#ff5733');
    } else {
      console.warn('Color picker not found. Please check if the ChromePicker renders correctly.');
    }
  });

  test('calls the onClose function when the Cancel button is clicked', () => {
    const cancelButton = screen.getByRole('button', { name: /Cancel/i });
    fireEvent.click(cancelButton);
    expect(mockOnClose).toHaveBeenCalled();
  });

  test('calls the addTextLayer function when the Add Text button is clicked', () => {
    const addButton = screen.getByRole('button', { name: /Add Text/i });
    fireEvent.click(addButton);
    expect(mockAddTextLayer).toHaveBeenCalled();
  });

  test('matches the snapshot of the component', () => {
    const { asFragment } = render(
      <TextLayerDialog
        open={true}
        onClose={mockOnClose}
        textInput=""
        setTextInput={mockSetTextInput}
        textFont="Arial"
        setTextFont={mockSetTextFont}
        textSize={24}
        setTextSize={mockSetTextSize}
        textColor="#000000"
        setTextColor={mockSetTextColor}
        addTextLayer={mockAddTextLayer}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
