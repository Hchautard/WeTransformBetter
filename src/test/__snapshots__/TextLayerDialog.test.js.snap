// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLayerDialog Component matches the snapshot of the component 1`] = `<DocumentFragment />`;
